lib/Common/Core/DbgHelpSymbolManager.cpp

```diff
@@ -53,7 +54,7 @@ DbgHelpSymbolManager::Initialize()
 
     // Let's make sure the directory where chakra.dll is, is on the symbol path.
 
-    char16 const * wszModule = AutoSystemInfo::GetJscriptDllFileName();
+    wszModule = AutoSystemInfo::GetJscriptDllFileName();
```

Comment:
The problem is not the `return`, it is just the `goto end.` that is skipping the declaration with an initializer.  Here is the applicable paragraph from the C++11 standard 
[stmt.dcl]/3
It is possible to transfer into a block, but not in a way that bypasses declarations with initialization. A program that jumps90 from a point where a variable with automatic storage duration is not in scope to a point where it is in scope is ill-formed unless the variable has scalar type, class type with a trivial default constructor and a trivial destructor, a cv-qualified version of one of these types, or an array of one of the preceding types **and is declared without an initializer** (8.6).